### ISSUE TITLE
Axis limits aligning with ticks

### ISSUE THREAD

**Issue by aahaselgrove** [JuliaPlots/Plots.jl#2155]
As discussed [here](https://discourse.julialang.org/t/plot-limit-at-next-tick-label/27704), the automatic choice of axis limits doesn't align with the automatic choice of axis ticks. I think the limits ending at an axis tick would be an improvement.

**Comment by daschw** [comment]
I'm not sure this is a good idea in general. I think it is rather up to personal taste if that is an improvement or not.

I think you know that you can provide ticks and axis limits manually to ensure this.
Alternatively you could also specifiy the number of desired ticks by passing an `Int` to ticks. In that case the axes limits always correspond to the last ticks:

```julia
plot(sin, -π, π, xticks = 5, yticks = 7)
```
![tickslimits](https://user-images.githubusercontent.com/16589944/63692350-bfd64280-c811-11e9-9af4-2b653b7e9d80.png)

**Comment by aahaselgrove** [comment]
Interesting, I didn't know that. Considering that's already behaviour, would it be an idea to add a kwarg for it, or some kind of default placeholder like 0 or -1? My plot axis ticks and limits depend on the data being displayed.

**Comment by wfrgra** [comment]
Also try 
`plot(sin, -π, π, lims=:round) `
This will expand the axis limits to round values and the tick values will be calculated from these limits
![sin](https://user-images.githubusercontent.com/28995447/66288698-ee256280-e925-11e9-8ab7-2162a510568c.png)

**Comment by aahaselgrove** [comment]
@wfrgra that seems to work to some extent, but possibly not for all magnitudes?
Eg.

````
using Plots
pyplot()

time = 0:20
spend = fill(2.1e6, 21)
bar(time, spend,
    title="Spend over time",
    xlabel="Year",
    ylabel="Spend",
    yformatter=y->"\$$y")
````
![Figure_1](https://user-images.githubusercontent.com/2556805/66290747-60e10e80-e923-11e9-9b40-909e5d9dde6e.png)

**Comment by wfrgra** [comment]
You're missing `lims=:round` from your command there

**Comment by aahaselgrove** [comment]
Good catch, but it wasn't missing from the actual command I ran, just from the version I pasted here.

**Comment by wfrgra** [comment]
Not sure what's going on there then. I don't have pyplot installed, but the limits are set independently of the backend so it shouldn't affect anything. With GR and your code above I get:

![bar](https://user-images.githubusercontent.com/28995447/66293521-8aedfd00-e932-11e9-93d3-d57d4f492a38.png)

**Comment by aahaselgrove** [comment]
My mistake. I have other plots which it doesn’t quite seem to be working for, trying to reproduce.

**Comment by aahaselgrove** [comment]
I've updated the example and believe it now demonstrates an issue.

**Comment by aahaselgrove** [comment]
Also, feature request: specify :round for one limit and a value for the other. Currently you can only get one or the other:
````
plot(sin, -π, π, xlims=(0, Inf), lims=:round)
````
![Figure_1](https://user-images.githubusercontent.com/2556805/66315482-3f9b1500-e959-11e9-96b8-9cf34c0da69c.png)
````
plot(sin, -π, π, lims=:round, xlims=(0, Inf))
````
![Figure_2](https://user-images.githubusercontent.com/2556805/66315489-432e9c00-e959-11e9-82e6-ed871f0e7692.png)

**Comment by wfrgra** [comment]
You're right, the axis doesn't always round to a number that the tick generating algorithm likes. I'll have a go at improving the match and see about allowing an override of one limit
